gui/mozregui/addons_editor.py

```diff
@@ -0,0 +1,89 @@
+from PyQt4.QtCore import QAbstractListModel, QModelIndex, Qt, \
+    pyqtSlot as Slot
+from PyQt4.QtGui import QWidget, QFileDialog
+from mozregui.ui.addons_editor import Ui_AddonsEditor
+
+
+class AddonsMode(QAbstractListModel):
+    """
+    A Qt model that can edit addons path.
+    """
+
+    def __init__(self, parent=None):
+        QAbstractListModel.__init__(self, parent)
+        self.addonsPath = []
+
+    def rowCount(self, index=QModelIndex()):
+        return len(self.addonsPath)
+
+    def data(self, index, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole:
+            if index.row() == 0:
+                return None
+            else:
+                return repr(self.addonsPath[index.row()])
+
+    def setData(self, index, new_value, role=Qt.EditRole):
+        path = str(new_value.toString())
+        self.addonsPath[index.row()] = path
+        return True
+
+    def headerData(self, section, orientation, role=Qt.DisplayRole):
+        if role == Qt.DisplayRole and orientation == Qt.Horizontal:
+            return 'path'[section]
+
+    def flags(self, index):
+        return Qt.ItemIsSelectable | Qt.ItemIsEditable | Qt.ItemIsEnabled
+
+    def add_addons_from_file(self, addons):
+        if addons:
+            path = len(self.addonsPath)
+            self.beginInsertRows(QModelIndex(), path,
+                                 path + len(addons) - 1)
+            self.addonsPath.extend(str(addons))
```

Comment:
I think you just need to use 'append' instead of 'extend'. But I think you misunderstood what a addon is. An addon for firefox is an extension (I think you new that but let's be clear), https://addons.mozilla.org/en-US/firefox/. Here we should give a list of addons, so I would say a list of file end each file name ends with '*.xpi'.

I think that the Profile class we use can also take a folder name that contains addons - but I am not sure, and this won't do us any good from the gui. I think we should just allow the user to select one or more *.xpi files.

Does that help ?